opacus/privacy_engine.py

```diff
@@ -493,3 +494,50 @@ def get_epsilon(self, delta):
             Privacy budget (epsilon) expended so far.
         """
         return self.accountant.get_epsilon(delta)
+
+    def save_checkpoint(
+        self,
+        *,
+        path: Union[str, os.PathLike, BinaryIO, IO[bytes]],
+        module: GradSampleModule,
+        optimizer: DPOptimizer,
```

Comment:
Should we maybe make optimizer optional here? 

1) You actually only need to save/load optimizer state if you're using parameter scheduling. Otherwise it should be enough to save model and the accountant.

2) Talking about schedulers, what's the recommended way to save/load if you're using noise scheduler? Do we want people to save it by themselves or through privacy engine too?

Reply:
Re 1: Point well taken.
Re 2: Scheduler already has a `state_dict()` method. I think it definitely makes sense to save that from Privacy Engine too.